generativeai/src/main/java/com/google/ai/client/generativeai/type/RequestOptions.kt

```diff
@@ -27,14 +27,21 @@ import kotlin.time.toDuration
  * @property timeout the maximum amount of time for a request to take, from the first request to
  *   first response.
  * @property apiVersion the api endpoint to call.
+ * @property autoFunction if false, auto functions will not be automatically executed
  */
-class RequestOptions(val timeout: Duration, val apiVersion: String = "v1") {
+class RequestOptions(
+  val timeout: Duration,
+  val apiVersion: String = "v1",
+  val autoFunction: Boolean = true
```

Comment:
Having a boolean default to `true` makes it seem like we're enforcing some type of behavior that should be optional to the developer. As a developer, I usually prefer that booleans default to `false` and I'll flip it to true if I need to change some behavior. (see similar threads on [StackOverflow](https://stackoverflow.com/a/14157372/5861618) and [YAQS (googlers-only)](https://yaqs.corp.google.com/eng/q/7514440696221138944#a1))

I think if we really want auto function to be the default behavior, we should consider changing the name of the flag so that it can default to false:
```suggestion
  val disableAutoFunction: Boolean = false
```